frontend/src/lib/components/ObjectTags/ObjectTags.tsx

```diff
@@ -49,14 +50,10 @@ export function ObjectTags({
     className,
     'data-attr': dataAttr,
 }: ObjectTagsProps): JSX.Element {
-    const [addingNewTag, setAddingNewTag] = useState(false)
-    const [newTag, setNewTag] = useState('')
-    const [deletedTags, setDeletedTags] = useState<string[]>([]) // we use this state var to remove items immediately from UI while API requests are processed
-
-    const handleDelete = (tag: string, currentTags?: string[], propertyId?: string): void => {
-        setDeletedTags([...deletedTags, tag])
-        onTagDelete && onTagDelete(tag, currentTags, propertyId)
-    }
+    const objectTagId = useMemo(() => uniqueMemoizedIndex++, [tags])
+    const logic = objectTagsLogic({ id: objectTagId, onChange, tags })
```

Comment:
This will recreate a new logic every time the `tags` array changes. I guess that works, but this is kind of unexpected, could be flakey and not very optimal as a lot of actions will be fired (logic unmounted, logic mounted, etc). I'd keep the logic intact for the entire duration of the widget, and use `useEffect(() => setTags(tags), [tags])` or something like that to bridge the gap from React to Kea.